File: XX - docs/selenium-webdriver/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 3/Java/ImplicitWaitTest.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;

import org.junit.Test;
import static org.junit.Assert.*;

import java.util.concurrent.TimeUnit;

public class ImplicitWaitTest {
 	@Test
 	public void testWithOutWait()
 	{
 		//Go to the Demo AjAX Application
 		WebDriver driver = new FirefoxDriver();
        driver.get("http://dl.dropbox.com/u/55228056/AjaxDemo.html");
        
        try {
        	
        	//Get link for Page 4 and click on it
        	WebElement page4button = driver.findElement(By.linkText("Page 4"));
        	page4button.click();
        	
        	//Get an element with id page4 and verify it's text
        	//Test will fail to locate the element without an Implicit Wait
        	WebElement message = driver.findElement(By.id("page4"));
           	assertTrue(message.getText().contains("Nunc nibh tortor"));
        	
        } catch (NoSuchElementException e) {
        	fail("Element not found!!");
        	e.printStackTrace();
        } finally {
        	driver.close();
        }
 	}
 	
 	@Test
 	public void testWithImplicitWait()
 	{
 		//Go to the Demo AjAX Application
 		WebDriver driver = new FirefoxDriver();
        driver.get("http://dl.dropbox.com/u/55228056/AjaxDemo.html");
        
        //Set the Implicit Wait time Out to 10 Seconds
 		driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
 		
 		try {

 			//Get link for Page 4 and click on it
 			WebElement page4button = driver.findElement(By.linkText("Page 4"));
 			page4button.click();
 		
 			//Get an element with id page4 and verify it's text
 			WebElement message = driver.findElement(By.id("page4"));
 			assertTrue(message.getText().contains("Nunc nibh tortor"));
 			
 		} catch (NoSuchElementException e) {
 			fail("Element not found!!");
 			e.printStackTrace();
 		} finally {
 			driver.close();
 		}
 	}
}

File: XX - Backup Project/xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 3/DotNet/ControllingTestFlowExamples/ControllingTestFlowExamples/ImplicitWaitTest.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using NUnit.Framework;

namespace ControllingTestFlowExamples
{
    [TestFixture]
    public class ImplicitWaitTest
    {
        [TestCase]
 	    public void TestWithImplicitWait()
 	    {
 		    //Go to the Demo AjAX Application
            IWebDriver driver = new ChromeDriver(@"C:\ChromeDriver");
            driver.Navigate().GoToUrl("http://dl.dropbox.com/u/55228056/AjaxDemo.html");
        
            //Set the Implicit Wait time Out to 10 Seconds
            driver.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(10));
 		
 		    try {

 			    //Get link for Page 4 and click on it
 			    IWebElement page4button = driver.FindElement(By.LinkText("Page 4"));
 			    page4button.Click();
 		
 			    //Get an element with id page4 and verify it's text
 			    IWebElement message = driver.FindElement(By.Id("page4"));
 			    Assert.IsTrue(message.Text.Contains("Nunc nibh tortor"));
 		    } 
            catch (NoSuchElementException e)
            {
 			    Assert.Fail("Element not found!!");
 		    } 
            finally 
            {
                driver.Close();
 		    }
 	    }
    }
}
